### PR TITLE
Move config hash stored in pod labels to pod annotations

### LINKED ISSUE #5071: Move labels used to store hashes in annotations
In order to reduce the number of labels added to the pods by ECK, there are labels used to store hashes that could be moved into annotations:
- `elasticsearch.k8s.elastic.co/config-hash`
- `elasticsearch.k8s.elastic.co/monitoring-config-hash`
- `elasticsearch.k8s.elastic.co/secure-settings-hash`
- `kibana.k8s.elastic.co/config-checksum`
- `kibana.k8s.elastic.co/monitoring-config-hash`
- `beat.k8s.elastic.co/config-checksum`
- `agent.k8s.elastic.co/config-checksum`
- `enterprisesearch.k8s.elastic.co/config-hash`
- `maps.k8s.elastic.co/config-hash`

Relates to #4617.

### PR DESCRIPTION
**Pull request by thbkrkr**
- Moves config hash stored in pod labels to pod annotations
- Removes ES secure settings hash and writes in the config hash instead
- Uses 32 bit FNV-1a hash algo most efficient than SHA224. It is already used for ES.

Resolves #5071.

This triggers a rollling update of all pods.

Tested: upgrade ECK from 1.9.0 to this PR :heavy_check_mark:

### CODE REVIEW

File: pkg/controller/common/hash/hash.go [hunk 1]
```diff
@@ -36,21 +37,27 @@ func GetTemplateHashLabel(labels map[string]string) string {
...
-// ensuring the hash does not change when a pointer changes.
-// The returned hash can be used for object comparisons.
-//
+// HashObject returns a hash of a given object using the 32-bit FNV-1 hash function
+// and the spew library to print the object (see WriteHashObject).
 // This is inspired by controller revisions in StatefulSets:
 // https://github.com/kubernetes/kubernetes/blob/8de1569ddae62e8fab559fe6bd210a5d6100a277/pkg/controller/history/controller_history.go#L89-L101
 func HashObject(object interface{}) string {
-	hf := fnv.New32()
+	objHash := fnv.New32()
```
Comment (pebrc) [inline pkg/controller/common/hash/hash.go:45]: any particular reason not to use FNV-1a which has slightly better characteristics

File: pkg/controller/elasticsearch/nodespec/podspec_test.go [hunk 2]
```diff
@@ -322,25 +321,24 @@ func TestBuildPodTemplateSpec(t *testing.T) {
...
 	tests := []struct {
-		name             string
-		args             args
-		expectedLabels   map[string]string
-		unexpectedLabels []string
-		wantErr          bool
+		name                  string
+		args                  args
+		expectedAnnotations   map[string]string
+		unexpectedAnnotations []string
```
Comment (pebrc) [inline pkg/controller/elasticsearch/nodespec/podspec_test.go:334]: Unexpected annotations is no longer used?

File: pkg/controller/elasticsearch/nodespec/podspec.go [hunk 3]
```diff
@@ -5,17 +5,17 @@
...
 
 import (
-	"crypto/sha256"
 	"fmt"
 	"hash/fnv"
 	"strings"
 
 	corev1 "k8s.io/api/core/v1"
 
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/hash"
```
Comment (pebrc) [inline pkg/controller/elasticsearch/nodespec/podspec.go:14]: import grouping

File: pkg/controller/elasticsearch/nodespec/podspec.go [hunk 4]
```diff
@@ -148,31 +151,43 @@ func buildLabels(
 	if err != nil {
```
Comment (pebrc) [inline pkg/controller/elasticsearch/nodespec/podspec.go:151]: Comment on line 146/149 is outdated

**Review by pebrc — COMMENTED** [review]
Looks good to me. I am slightly wary of the rolling restart this causes. But I guess handling labels/annotations as a special case that does not go through the regular reconciliation would probably be too much of a mess.

File: pkg/controller/common/hash/hash.go (hunk 1 above)
Comment (thbkrkr) [reply pkg/controller/common/hash/hash.go]: No, I've updated to use FNV-1a.

**Review by pebrc — APPROVED** [review]
LGTM

**Pull request merged**